File: Bachelor/Semester5/Mobile_Application_Programming/server/Services/src/main/java/service/Service.java
package service;

import model.Expense;

import java.util.List;

public interface Service {
    Expense addExpense(Expense expense);

    Expense deleteExpense(String id);

    Expense updateExpense(Expense expense);

    List<Expense> getAllExpenses();

    Expense findExpense(String id);
}

File: Bachelor/Semester5/Mobile_Application_Programming/android/testmergeandroid/app/src/main/java/service/Service.kt
package service

import model.Expense
import repository.Repository
import java.io.Serializable
import java.util.*

interface Service : Serializable {
    fun findAllExpenses(): List<Expense>

    fun addExpense(e: Expense): Optional<Expense>

    fun deleteExpense(id: String): Optional<Expense>

    fun updateExpense(e: Expense): Optional<Expense>

    fun findExpenseById(id: String): Optional<Expense>
}